mne/io/neuralynx/neuralynx.py

```diff
@@ -0,0 +1,159 @@
+        assert sum([len(segment.analogsignals) for segment in bl[0].segments]) == len(
+            bl[0].segments
+        )
+
+        # just a wrapper for readability
+        def _find_first_last_segment(
+            segments, start_spl: int, stop_spl: int, sr: float
+        ):
+            # gather the start and stop times (in sec) for all signals in segments (assumes 1 signal per segment)
+            # shape = (n_segments, 2) where 2nd dim is (start_time, stop_time)
+            onst_offt = np.array(
+                [
+                    (signal.t_start.item(), signal.t_stop.item())
+                    for segment in segments
+                    for signal in segment.analogsignals
+                ]
+            )
+
+            # if start/stop are not speficied, read all segments
+            start_segment = 0
+            stop_segment = len(segments) - 1  # -1 because python indexing
+
+            # determine which segment contains the start sample and which the stop sample if start/stop is specified
+            if start_spl > 0:
+                start_time = start_spl / sr
+                start_segment = np.where(
+                    (onst_offt[:, 0] <= start_time) & (onst_offt[:, 1] >= start_time)
+                )[0].item()
+            if stop_spl is not None:
+                stop_time = stop_spl / sr
+                stop_segment = np.where(
+                    (onst_offt[:, 0] <= stop_time) & (onst_offt[:, 1] >= stop_time)
+                )[0].item()
+
+            return start_segment, stop_segment, onst_offt
+
+        stop_spl = (
+            None if stop == data.shape[-1] else stop
+        )  # assume we read until the end if not speficied otherwise
+
+        first_seg, last_seg, start_stop_times = _find_first_last_segment(
+            bl[0].segments, start_spl=start, stop_spl=stop_spl, sr=sr
+        )
+
+        # now select only segments between the one that containst the start sample
+        # and the one that contains the stop sample
+        sel_times = start_stop_times[first_seg : last_seg + 1, :]
+
+        if (start is not None) or (start > 0):
+            sel_times[0, 0] = (
+                start / sr
+            )  # for the segment containting the start sample, reset the segment start time to the requested `start` value
+        if stop_spl is not None:
+            sel_times[-1, 1] = stop / sr  # same but for stop sample
+
+        # now load the data arrays via neo.Segment.Analogsignal.load() only from the selected segments and channels
+        all_data = np.concatenate(
+            [
+                signal.load(
+                    channel_indexes=idx, time_slice=(time[0], time[-1])
+                ).magnitude
+                for seg, time in zip(
+                    bl[0].segments[first_seg : last_seg + 1], sel_times
+                )
+                for signal in seg.analogsignals
+            ]
+        ).T
+
+        block = all_data  # shape = (len(idx), n_samples))
```

Comment:
Perhaps somewhat confusingly, whatever you pass to `_mult_cal_one` needs to be shape `(n_channels, n_samples)`, not `(len(idx), n_samples)`. I'll push a quick commit to show what I mean